cardano-node/src/Cardano/Tracing/OrphanInstances/Shelley.hs

```diff
@@ -389,16 +394,81 @@ instance (ShelleyBasedEra era, Core.Value era ~ Coin)
+             , "error" .= renderBadInputsUTxOErr badInputs
+             ]
+  toObject _verb (MA.OutsideValidityIntervalUTxO validityInterval slot) =
+    mkObject [ "kind" .= String "ExpiredUTxO"
+             , "validityInterval" .= validityInterval
+             , "slot" .= slot ]
+  toObject _verb (MA.MaxTxSizeUTxO txsize maxtxsize) =
+    mkObject [ "kind" .= String "MaxTxSizeUTxO"
+             , "size" .= txsize
+             , "maxSize" .= maxtxsize ]
+  toObject _verb MA.InputSetEmptyUTxO =
+    mkObject [ "kind" .= String "InputSetEmptyUTxO" ]
+  toObject _verb (MA.FeeTooSmallUTxO minfee txfee) =
+    mkObject [ "kind" .= String "FeeTooSmallUTxO"
+             , "minimum" .= minfee
+             , "fee" .= txfee ]
+  toObject _verb (MA.ValueNotConservedUTxO consumed produced) =
+    mkObject [ "kind" .= String "ValueNotConservedUTxO"
+             , "consumed" .= consumed
+             , "produced" .= produced
+             , "error" .= renderValueNotConservedErr consumed produced
+             ]
+  toObject _verb (MA.WrongNetwork network addrs) =
+    mkObject [ "kind" .= String "WrongNetwork"
+             , "network" .= network
+             , "addrs"   .= addrs
+             ]
+  toObject _verb (MA.WrongNetworkWithdrawal network addrs) =
+    mkObject [ "kind" .= String "WrongNetworkWithdrawal"
+             , "network" .= network
+             , "addrs"   .= addrs
+             ]
+  -- TODO: Add the minimum allowed UTxO value to OutputTooSmallUTxO
+  toObject _verb (MA.OutputTooSmallUTxO badOutputs) =
+    mkObject [ "kind" .= String "OutputTooSmallUTxO"
+             , "outputs" .= badOutputs
+             , "error" .= String "The output is smaller than the allow minimum \
+                                 \UTxO value defined in the protocol parameters"
+             ]
+  toObject verb (MA.UpdateFailure f) = toObject verb f
+  toObject _verb (MA.OutputBootAddrAttrsTooBig badOutputs) =
+    mkObject [ "kind" .= String "OutputBootAddrAttrsTooBig"
+             , "outputs" .= badOutputs
+             , "error" .= String "The Byron address attributes are too big"
+             ]
+  toObject _verb MA.TriesToForgeADA =
+    mkObject [ "kind" .= String "TriesToForgeADA" ]
+
 renderBadInputsUTxOErr ::  Set (TxIn era) -> Value
 renderBadInputsUTxOErr txIns
   | Set.null txIns = String "The transaction contains no inputs."
   | otherwise = String "The transaction contains inputs that do not exist in the UTxO set."
 
-renderValueNotConservedErr :: DeltaCoin -> DeltaCoin -> Value
-renderValueNotConservedErr consumed produced
-  | consumed > produced = String "This transaction has consumed more Lovelace than it has produced."
-  | consumed < produced = String "This transaction has produced more Lovelace than it has consumed."
-  | otherwise = String "Impossible: Somehow this error has occurred in spite of the transaction being balanced."
+-- TODO @node team: does this make sense?
+renderValueNotConservedErr :: Show val => val -> val -> Value
```

Comment:
We use `renderValueNotConservedErr` with two instantiations of `val`:
* [Shelley and Allegra](https://github.com/input-output-hk/cardano-ledger-specs/blob/f1b1a0c3832aa3edffecebd744acd978d5d6611e/shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Utxo.hs#L137-L139): `Delta (Core.Value era)`, which reduces to `DeltaCoin`
* [Mary](https://github.com/input-output-hk/cardano-ledger-specs/blob/e1f839bff708e1938355b56ad3a7fe03930851ff/shelley-ma/impl/src/Cardano/Ledger/ShelleyMA/Rules/Utxo.hs#L95-L97): `Core.Value era`, which reduces to [`Value`](https://github.com/input-output-hk/cardano-ledger-specs/blob/e1f839bff708e1938355b56ad3a7fe03930851ff/shelley-ma/impl/src/Cardano/Ledger/Mary/Value.hs#L75).

By specialising to the concrete types you could recover the more detailed error message.